Refactor huawei cloudeye scaler

rickbrouwer opened the pull request:
Refactor the scaler to match the new metadata guidelines.

Good review required, because there is no e2e test for the Huawei Cloudeye scaler.

Because there is a deprecate announcement in it I will create a changelog item and the items according the [deprecation policy.](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)

### Checklist

- [X] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [X] Changelog has been updated and is aligned with our [changelog requirements]
- [X] Tests have been added
- [X] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [X] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Relates to #5797
Docs: https://github.com/kedacore/keda-docs/pull/1610

JorTurFer reviewed (approved):
I've checked the parameters one by one and I don't see any typo 😄

JorTurFer commented:
/skip-e2e

rickbrouwer commented:
Hi @wozniakjan 

Can you add a deprecation announcement on [GitHub Discussions](https://github.com/kedacore/keda/discussions/categories/deprecations)

I'm not allowed to do that yet.

rickbrouwer commented:
> I've updated the role but I'm not sure if this permissions will be enough (I've added all permissions related with discussions, so I hope it so), have a try and I'll open it if you still can't

Unfortunately, I don't get the option

JorTurFer commented:
The it looks that it's not possible if the person doesn't have admin/maintainer permission explicitly set 😢 
I'm going to create it right now

JorTurFer commented:
https://github.com/kedacore/keda/discussions/7014
